nearcore/src/config.rs

```diff
@@ -1108,20 +1112,223 @@ pub enum FileDownloadError {
     RemoveTemporaryFileError(std::io::Error, Box<FileDownloadError>),
 }
 
+/// Object which allows transparent XZ decoding when saving data to a file.
```

Comment:
optional: I'd put this whole struct in a separate file.

Reply:
Yes, I’ve been thinking about that. It seems that we need an additional crate with just utility functions and classes which aren’t specific to any other NEAR thing.